### ISSUE TITLE
CVodeFree has correct type but gets called incorrectly

### ISSUE THREAD

**Issue by ikirill** [JuliaLang/Sundials.jl#48]
`CVodeFree` takes a pointer to a `void* mem` object from `CVodeCreate`, not the object itself.

So the correct call is

    ccall((:CVodeFree, Sundials.libsundials_cvodes), Void, (Ptr{Ptr{Void}},), &mem)

not as it is called now if you try to pass `mem` to it:

    CVodeFree(mem)
    ==> ccall((:CVodeFree, Sundials.libsundials_cvodes), Void, (Ptr{Ptr{Void}},), mem)

This bug leads to unexpected segfaults when freeing resources. `CVodeFree` looks like this:

    SUNDIALS_EXPORT void CVodeFree(void **cvode_mem);

**Comment by ViralBShah** [comment]
I just gave you commit access. Could you submit PRs for this and the other issues you have filed?

**Comment by ViralBShah** [comment]
Could you also review the outstanding PRs?

**Comment by christianhaargaard** [comment]
I tested the call above with a customized version of the code posted by rs1909 in #24, and it works great. How difficult is it to fix the CVodeFree call?

**Comment by acroy** [comment]
@christianhaargaard : I think you can just call it like that
```
cvode_mem = Sundials.CVodeCreate(Sundials.CV_BDF, Sundials.CV_NEWTON)
...
Sundials.CVodeFree([cvode_mem])
```
At least this is the approach taken in #29, which is/was supposed to fix #24.

**Comment by christianhaargaard** [comment]
Thanks! I was getting a little sad seeing as I have to solve my ODE a couple of hundred times :-)

**Comment by acroy** [comment]
Should be fixed by #57. Please reopen if you this is still a problem.
